types/plugin.d.ts

```diff
@@ -1,19 +1,30 @@
 import { FastifyInstance } from './instance'
-import { FastifyError } from 'fastify-error'
-import { RawServerBase, RawServerDefault, RawRequestDefaultExpression, RawReplyDefaultExpression } from './utils'
+import {
+  RawServerBase,
+  RawRequestDefaultExpression,
+  RawReplyDefaultExpression
+} from './utils'
 
 /**
- * FastifyPlugin
+ * FastifyPluginCallback
  *
  * Fastify allows the user to extend its functionalities with plugins. A plugin can be a set of routes, a server decorator or whatever. To activate plugins, use the `fastify.register()` method.
  */
-export interface FastifyPlugin<Options extends FastifyPluginOptions = {}> {
-  (
-    instance: FastifyInstance<RawServerBase, RawRequestDefaultExpression<RawServerBase>, RawReplyDefaultExpression<RawServerBase>>,
-    opts: Options,
-    next: (err?: FastifyError) => void
-  ): void;
-}
+export type FastifyPluginCallback<Options extends FastifyPluginOptions = {}> = (
```

Comment:
I think such definition will break libraries such as `fastify-autoload` which rely upon existence of unified FastifyPlugin type (https://github.com/fastify/fastify-autoload/blob/master/fastify-autoload.d.ts) 
I guess they will have to also use overload if there is no other way.

Reply:
So all the plugins types would need to be updated _again_? :/

Reply:
We have a few options:
- see if the union or intersection type will satisfy existing plugins
- fix all plugins again (☹️)
- add an alias for FastifyPlugin = FastifyPluginCallback as this is what it was before the change; mark as deprecated and then expect that update in v4

Reply:
let's go with the 3rd route.

Reply:
Wouldn't union plugin type work out just fine for plugins, everything will work as it was but it will allow passing explicitly typed async functions to plugins (it just won't support type inference for async functions)?

Reply:
I'm not sure off the top of my head but please try it out